detectors/aws/lambda/detector.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lambda
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+const (
+	lambdaFunctionNameEnvVar    = "AWS_LAMBDA_FUNCTION_NAME"
```

Comment:
are any of the other variables valuable here? if not maybe a note that points a user to where they can find the list would be useful: https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html

Reply:
None of the other environment variables store data that should be recorded per the [faas spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/faas.md), but I'll make a suggestion to include that link in a comment